docs/02-app/01-building-your-application/04-caching/index.mdx

```diff
@@ -0,0 +1,421 @@
+---
+title: Caching in Next.js
+nav_title: Caching
+description: An overview of caching mechanisms in Next.js.
+---
+
+Next.js has a few caching mechanisms to improve your application's performance. This page provides an in-depth look at caching, the APIs you can use to configure them, and how they interact with each other.
+
+## Overview
+
+{/* TO DO: Diagram showing how caching mechanisms come into play in the request lifecycle, going from initial request, to server, to response, to router cache, to subsequent navigation with new request. */}
+
+Here's a high-level overview of the different caching mechanisms in Next.js and their purpose:
+
+| Mechanism                     | What is cached?                | Where is it cached? | Why is it cached?                                             | Duration                        |
+| ----------------------------- | ------------------------------ | ------------------- | ------------------------------------------------------------- | ------------------------------- |
+| [React Cache](#react-cache)   | Return values of functions     | Server              | Reduce data store calls and share data between route segments | Per-request lifecycle           |
+| [Data Cache](#data-cache)     | Return values of data requests | Server              | Store data across user requests and deployments               | Persistent (can be revalidated) |
+| [Render Cache](#render-cache) | Rendered React Components      | Server              | Reduce rendering cost and improve performance                 | Persistent (can be revalidated) |
+| [Router Cache](#router-cache) | Route Segments (RSC Payload)   | Client              | Reduce server requests and improve navigation experience      | User session or time-based.     |
+
+## How caching works
+
+### React Cache
+
+React's `cache` function allows you to **memoize** the return value of a function, saving the result to memory. This means you can call the same function multiple times while only executing it once during a render pass.
+
+In Next.js, the [`fetch` API](#fetch) uses `cache` to **deduplicate** requests for the same data. Similar to memoization, request deduplication is an optimization that takes **multiple** requests for the same data and only makes a **single** request to the data source. This is done by storing the result of the first request in a temporary server-side cache, and returning the cached result for subsequent requests.
+
+{/* Diagram */}
+
+Since `fetch` requests are automatically deduplicated, you can fetch data in the components that need it without props drilling and worrying about the performance impact of making multiple requests for the same data. Deduplication applies to `fetch` requests in `generateMetadata`, `generateStaticParams`, Layouts, Pages, and other Server Components.
```

Comment:
```suggestion
Since `fetch` requests are automatically memoized, you can fetch data in the components that need it without props drilling and worrying about the performance impact of making multiple requests for the same data. Memoization applies to `fetch` requests in `generateMetadata`, `generateStaticParams`, Layouts, Pages, and other Server Components.
```